src/neural/loader.cc

```diff
@@ -140,6 +209,37 @@ void FixOlderWeightsFile(WeightsFile* file) {
       net->set_smolgen_activation(pblczero::NetworkFormat::ACTIVATION_SWISH);
     }
   }
+
+  // Get updated network format.
+  network_format = file->format().network_format().network();
+  auto embedding_type = file->format().network_format().input_embedding();
+  bool multihead_format = (network_format & 128) == 128;
+  if (!multihead_format && network_format != nf::NETWORK_ONNX) {
```

Comment:
Maybe, to reduce nesting...

```cpp
if (network_format == nf::NETWORK_ONNX) return;

if (multihead_format) {
   // ...
   return;
}

if (weights.has_policy_heads() && weights.has_value_heads()) {
   // ...
   return;
}

// the rest.
```

Reply:
This will likely be replaced by the code in #1915 (but still under revision)